리스트페이지를 구성한다.

GyeongyeonAn opened the pull request:
![image](https://user-images.githubusercontent.com/55589616/119818237-ea46b980-bf29-11eb-9a54-dd35b749a347.png)


임시 데이터 셋으로 목록을 구성 완료했습니다!
피드백 부탁드립니다. 

추후에 모달이 완성되면 이슈 하나 올려서 필터 적용과 같이 할게요.

GyeongyeonAn committed:
chore: divide section(filter, section) and pull data image

GyeongyeonAn committed:
style: 리스트 요소 적절한 위치로 배치

GyeongyeonAn committed:
style: 필터 추가 and 스크롤 추가(리스트-세로, 필터-세로)

GyeongyeonAn committed:
feat: 리스트 항목 클릭시 이벤트 발생 기능 추가

GyeongyeonAn committed:
refactor: 필터를 공통 컴포넌트로 분리

hwookim commented on src/pages/list/index.jsx:
```diff
@@ -1,11 +1,102 @@
+  border-bottom: 1px solid #dbdbdb;
+`;
+
+const Thumb = styled.div`
```
엄지인가요?

image에서 높이를 지정해준다면 아예 필요 없는 친구네요.

GyeongyeonAn replied:
넵 지웠습니다!

hwookim commented on src/pages/list/index.jsx:
```diff
@@ -1,11 +1,102 @@
 export default function ListPage() {
+  const listItems = places.map((post) => (
+    <ListItme key={post.name}>
+      <a onClick={() => alert("1")}>
```
왜 있는 코드죠?
불필요한 코드는 최대한 없애주세요.

실제로 모달창을 만들어도 a태그는 쓰이지 않을겁니다.

GyeongyeonAn replied:
넵 불필요한 코드는 안만들도록 하겠습니다

hwookim commented on src/pages/list/index.jsx:
```diff
@@ -1,11 +1,102 @@
+`;
 
 export default function ListPage() {
+  const listItems = places.map((post) => (
```
이걸 굳이 변수로 빼는 이유가 있나요?

분리를 할거라면 별도 컴포넌트로 하는 것이 나아보이는군요.
post를 props로 받는 컴포넌트를 만들고, 그걸 출력하는 것이 더 좋아보여요.

GyeongyeonAn replied:
![image](https://user-images.githubusercontent.com/55589616/119910739-c075ad00-bf92-11eb-913e-2d196c37cca4.png)

![image](https://user-images.githubusercontent.com/55589616/119910758-ce2b3280-bf92-11eb-8e4f-adde35a9ae62.png)

다음과 같이 수정했습니다!

변경된 네이밍
post -> place

hwookim replied:
한 파일 내에 같이 있는다면 이전이랑 뭐가 달라진걸까요.
별도의 컴포넌트로 구분하는 게 낫지 않을까요?

컴포넌트임을 나타내시려면 `<ListItem place={place} />` 와 같은 식으로 표현해주세요.
해당 컴포넌트 자체는 하나의 아이템이기 때문에 `ListItem`이 맞습니다.

hwookim commented on src/pages/recommend/index.jsx:
```diff
@@ -1,11 +1,30 @@
+import styled from "styled-components";
 import Header from "../../components/Header";
 
+const Container = styled.div`
```
커밋을 그대로 끌어오게 되면,
본인이 수정하지 않았을 코드들이 리뷰에 포함되면서 리뷰어에게 혼란을 주게 됩니다.

우리야 물론 단 3명이고, 제가 모든 코드를 파악하고 있으니 상관 없다지만
사람이 많은 프로젝트일 수록 리뷰어의 부담이 가중되고 업무의 처리 속도도 늦어질거예요.
되도록이면 pull --rebase 기능을 쓰시길 바라요.

GyeongyeonAn replied:
네 명심하겠습니다.

hwookim commented on src/components/Filter.jsx:
```diff
@@ -0,0 +1,32 @@
+const FilterHeader = styled.section`
+  white-space: nowrap;
+  overflow-x: auto;
+  padding: 0 15px 10px;
+  border-bottom: 1px solid #dbdbdb;
+`;
+
+const Button = styled.button`
+  display: inline-block;
+  margin: 10px 8px 0 8px;
```
필터 내 여백을 필터의 padding으로 관리할지, Button의 margin으로 관리할지
한가지로 통일하면 좋겠어요.
Header 컴포넌트의 margin-bottom을 없애고 현재 컴포넌트에서 온전히 관리하는 것도 방법입니다.

GyeongyeonAn replied:
Header 컴포넌트의 margin-bottom을 없애는 것이 좋을 것 같아요!
필터 내 여백 기능은 충돌 방지를 위해 feature/26에서 수정하는 걸로 할게요!

hwookim replied:
?? 다른 사람 코드에서는 계속 이상한 채로 두겠네요..
그냥 Filter 자체를 삭제하는 게 나을 거 같은데요 그런 식이시라면

GyeongyeonAn replied:
삭제했습니다!

hwookim commented on src/components/Filter.jsx:
```diff
@@ -0,0 +1,32 @@
+const Button = styled.button`
+  display: inline-block;
+  margin: 10px 8px 0 8px;
+  padding: 0 20px;
+  font-size: 0.875rem;
+  line-height: 32px;
```
line-height는 선의 높이, 공백의 높이를 설정하는데요.
이를 통해 크기를 조절하려는 시도는 좀 부적절해보이네요.

height를 쓰시거나, 차라리 padding을 활용하는 게 나아보입니다.

GyeongyeonAn replied:
넵 feature/26에서 수정했습니다.

hwookim commented on src/pages/list/index.jsx:
```diff
@@ -1,11 +1,102 @@
+  overflow-x: hidden;
+`;
+
+const ListItme = styled.li`
```
```suggestion
const ListItem = styled.li`
```

hwookim commented on src/pages/list/index.jsx:
```diff
@@ -1,11 +1,102 @@
+`;
+
+const Image = styled.img`
+  position: absolute;
+  top: 0;
+  left: 0;
+  width: 100%;
+  height: 100%;
+  object-fit: cover;
+  color: rgba(255, 255, 255, 0);
+  text-indent: -9999px;
+  border: none;
+  vertical-align: top;
```
width, height, object-fit을 제외하곤 전부 필요없는 코드네요

hwookim commented on src/pages/list/index.jsx:
```diff
@@ -1,11 +1,102 @@
+  font-size: 0.75rem;
+  margin-top: 8px;
+  color: #9b9b9b;
+  line-height: 1.3em;
+  overflow: hidden;
+  text-overflow: ellipsis;
+  white-space: nowrap;
+  padding-right: 10px;
```
필요 없는 코드로 보입니다.

GyeongyeonAn replied:
정말 필요 없네요..
다른 플랫폼에서 copy해서 가져오니 이런 현상이 일어나는 것 같습니다..
다음에는 하나씩 검색해보고 지우면서 시도해볼게요.

hwookim replied:
적어도 자신이 복사하는 코드가 뭔지는 이해하시는 것이 좋을 것 같아요.

hwookim commented on src/pages/list/index.jsx:
```diff
@@ -1,11 +1,102 @@
+import styled from "styled-components";
+import places from "../../data/places";
+
+const Contents = styled.div`
+  width: 100%;
+  height: 100%;
+  overflow-y: auto;
+  overflow-x: hidden;
+`;
```
여기에 overflow를 조절해봤자.
App 컴포넌트에서 hidden으로 했기 때문에 결국 overflow-y가 잘리네요.

아예 필요 없는 컴포넌트로 보입니다.

GyeongyeonAn replied:
지워도 이상 없는 것 확인 했습니다!

질문:
1. 지금 y축으로 스크롤 기능은 어디에 코드가 쓰여있나요?
2. 아래 잘린거는 어떻게 해결하는게 좋을까요?

GyeongyeonAn replied:
APP에 코드를 직접 수정해서 해결했습니다. 괜찮은가요?
overflow: hidden; -> overflow-y: auto; overflow-x: hidden

![image](https://user-images.githubusercontent.com/55589616/119913854-f4a09c00-bf99-11eb-9ba7-28c54b51564a.png)

![image](https://user-images.githubusercontent.com/55589616/119914032-582ac980-bf9a-11eb-815a-22919bd8646a.png)

hwookim commented on src/pages/list/index.jsx:
```diff
@@ -1,11 +1,102 @@
+
+const Info = styled.div`
+  padding: 12px 7px 16px 0px;
+  position: relative;
+  text-decoration: none;
+  vertical-align: middle;
```
필요 없는 코드네요.

GyeongyeonAn replied:
그러네요, 검색해보니 뜬금이 없어서 웃음이 나올 정도네요 ㅠㅠ

hwookim commented on src/pages/list/index.jsx:
```diff
@@ -1,11 +1,102 @@
+
+const Score = styled.span`
+  color: #ff792a;
+  position: static;
```
필요 없는 코드네요

GyeongyeonAn replied:
기본 포지션이 static이네요. 지웠습니다!

hwookim commented on src/pages/list/index.jsx:
```diff
@@ -1,11 +1,102 @@
+const Review = styled.p`
+  margin-top: 10px;
+  font-size: 0.875rem;
+  line-height: 21px;
```
영향을 주긴 하지만... 필요한 코드인지 의문이네요.

GyeongyeonAn replied:
있으나 없으나 차이가 거의 없는 걸 보고 지웠습니다!

GyeongyeonAn replied:
음... 미세한 차이긴 한데 제 눈에는 있는게 더 이뻐보이긴 하네요 ㅎㅎ..

hwookim replied:
저는 line-height 하나만을 의미했던 겁니다.

하나 집으면 위에꺼까지 같이 나오는군요.
다음부턴 리뷰에 좀 더 명시하도록 할게요.

GyeongyeonAn replied:
line-height 영향이 미비해서 제거했습니다!

hwookim reviewed (changes requested):
리뷰에도 남겼지만,
차라리 머지를 했다면 file change가 이러지 않았을 것 같네요.

css 처리 중 불필요한 코드를 찾기 위해선, 하나씩 일단 지워보도록 하세요.
지워도 변화가 없으면 그냥 불필요한 코드니 과감하게 삭제하시면 됩니다.

스타일을 위한 변수를 선언할 때,
아래에서 쓰일 순서와 동일하게 해주시면 더 좋습니다.
`Title - Score - Address - Review` 순서대로 쓰인다면,
위쪽의 스타일 컴포넌트도 같은 순서로요.

모든 것을 하나의 컴포넌트 내에서 끝낼 필요는 없습니다.
필요하다면 ListItem이라는 컴포넌트를 만들어서 해당 컴포넌트에 정보를 넘기고 코드를 분할하는 편이 더 좋을 수도 있어요.
그러고보니 places를 map 하는데 post가 나오는건 좀 신기하군요.
post는 너무 뜬금없지 않나요?

네이밍에 있어서,
되도록이면 Image, Button 같은 태그 그대로의 이름 보다는 어떤 역할인지 좀 더 나타내는 편이 좋습니다.
이미지가 두개 들어가면 Image1, Image2를 하게 되는 참사가 벌어지니까요.
Thumbnail, FilterButton 과 같은 식으로 하는 네이밍이 더 구분되고 좋지 않을까요?

고생하셨습니다.

GyeongyeonAn committed:
refactor: 불필요한 코드 삭제
style: Header.jsx margin-bottom 삭제, App.jsx overflow y축 스크롤 추가

hwookim commented on src/pages/list/index.jsx:
```diff
@@ -1,11 +1,73 @@
+function ListItems(place) {
+  return (
+    <ListItme key={place.name}>
+      <Image src={place.photo} />
```
단순 이미지보단 Thumbnail이라는 네이밍이 더 낫지 않을까요?

GyeongyeonAn replied:
아 아까 말해주셨는데 반영을 안했네요 알겠습니다!

hwookim commented on src/pages/list/index.jsx:
```diff
@@ -1,11 +1,73 @@
+
+function ListItems(place) {
+  return (
+    <ListItme key={place.name}>
```
```suggestion
    <ListItem key={place.name}>
```

오타는 여전히 수정이 안되어있네요.
ListItem 을 별도의 컴포넌트로 아예 분리하고,
Container 등으로 스타일 컴포넌트를 바꾸는 것도 방법입니다.

GyeongyeonAn replied:
ListItem을 공통 컴포넌트로 만들고 Container로 감쌋습니다.

![image](https://user-images.githubusercontent.com/55589616/119927540-91703300-bfb4-11eb-9687-71c3cf55a33a.png)

![image](https://user-images.githubusercontent.com/55589616/119927558-97feaa80-bfb4-11eb-89e5-bb22faa2a980.png)

hwookim commented on src/pages/list/index.jsx:
```diff
@@ -1,11 +1,73 @@
+  border-bottom: 1px solid #dbdbdb;
+`;
+
+const Image = styled.img`
+  top: 0;
+  left: 0;
+  width: 100%;
+  height: 144px;
+  object-fit: cover;
+  color: rgba(255, 255, 255, 0);
+  text-indent: -9999px;
+  border: none;
+  vertical-align: top;
+`;
```
어떤 코드가 필요한 코드고 어떤 코드가 필요 없는 코드인지 한 번 더 확인해보시기 바랍니다.

GyeongyeonAn replied:
죄송합니다. 다음에는 더 주의깊게 보겠습니다.

hwookim commented on src/components/Filter.jsx:
```diff
@@ -0,0 +1,32 @@
+import React from "react";
+import styled from "styled-components";
+
+const FilterHeader = styled.section`
+  white-space: nowrap;
+  overflow-x: auto;
+  padding: 0 15px 10px;
+  border-bottom: 1px solid #dbdbdb;
+`;
```
white-space는 왜 넣으신 걸까요?

GyeongyeonAn replied:
필터의 개수가 많아질 경우 

![image](https://user-images.githubusercontent.com/55589616/119928361-64bd1b00-bfb6-11eb-87cd-715f2ced3e6c.png)

이런 느낌으로 가로로 이어지고 스크롤로 필터들을 볼 수 있도록 하려고 했는데, 필터가 3개니 필요가 없겠네요. 삭제하겠습니다.

hwookim replied:
해당 요소는 말씀하신 방향이랑 전혀 관계가 없어보이는데요.
white-space: nowrap은 텍스트 내부 공백에 관한 설정입니다.

GyeongyeonAn committed:
refactor: 코드 개선
- Image <- Thumnail 네이밍 변경
- 불필요한 스타일 코드 제거
- ListItem 공통 컴포넌트로 분리
- Filter 삭제 (feature/26에서 개발)

hwookim commented on src/pages/list/index.jsx:
```diff
@@ -1,11 +1,13 @@
     <>
       <Header />
-      <div>List</div>
+      <ul>{places.map((place) => ListItem(place))}</ul>
```
```suggestion
      <ul>{places.map((place) => (<ListItem place={place} />))}</ul>
```

컴포넌트는 컴포넌트답게 표시하는 게 좋지 않을까요.

그 전에 우선 ListItem `함수`를 `컴포넌트`로 바꾸시고요

hwookim commented on src/components/ListItem.jsx:
```diff
@@ -0,0 +1,52 @@
+        <Review>{place.comment}</Review>
+      </Info>
+    </Container>
+  );
```
```suggestion
export default function ListItem({ place }) {
  return (
    <Container key={place.name}>
      <Thumbnail src={place.photo} />
      <Info>
        <Title>{place.name}</Title>
        <Score>{place.score}/5</Score>
        <Address>{place.address}</Address>
        <Review>{place.comment}</Review>
      </Info>
    </Container>
  );
```

컴포넌트라면 [구조분해할당](https://ko.javascript.info/destructuring-assignment)을 통해 props를 받아야합니다.
저 상태라면 그냥 함수지 컴포넌트가 아녜요.

win9612 commented:
코드적인 부분에서는 앞서 얘기가 다 된것 같고, 디자인 부분에 대해서 하나 의견만 내자면 별점 앞에 특수문자 ★을 다는 것이 시각적으로 좋아 보입니다. 그리고 음식점 이름은 Bold로 해서 다른 텍스트보다 눈에 띄면 좋을것 같고요.

GyeongyeonAn commented:
> 코드적인 부분에서는 앞서 얘기가 다 된것 같고, 디자인 부분에 대해서 하나 의견만 내자면 별점 앞에 특수문자 ★을 다는 것이 시각적으로 좋아 보입니다. 그리고 음식점 이름은 Bold로 해서 다른 텍스트보다 눈에 띄면 좋을것 같고요.

![image](https://user-images.githubusercontent.com/55589616/119940932-df903100-bfca-11eb-8d02-9042a8bd5b00.png)

GyeongyeonAn committed:
refactor: 피드백 적용하여 코드 개선

GyeongyeonAn committed:
refactor: ListItem을 list 위치로 이동

hwookim reviewed (approved):
제 꼬장 때문에 고생이 많으십니다...ㅎ헣헣ㅎㅎ

아닙니다 감사함다 ㅎㅎ
